src/pages/BuildListPage/BuildListPage.js

```diff
@@ -0,0 +1,143 @@
+import BranchSelect from '../../components/BranchSelect/BranchSelect';
+
+import './BuildListPage.css';
+
+import {
+  getBranchBuildList,
+  getProjectBuildList,
+  getBuildList,
+  selectBuild,
+} from '../../reducers/BuildReducer/actions';
+
+class BuildListPage extends Component {
+  componentDidMount() {
+    this.fetchBuilds(this.props.project, this.props.branch);
+  }
+
+  fetchBuilds = (project, branch) => {
+    if (project && branch) {
+      this.props.getBranchBuildList(this.props.apiKey, project, branch);
+    } else if (project) {
+      this.props.getProjectBuildList(this.props.apiKey, project);
+    } else {
+      this.props.getBuildList(this.props.apiKey);
+    }
+  }
+
+  selectBuild = (build) => () => {
+    this.props.selectBuild(build);
+  }
+
+  selectProject = (project) => {
+    this.fetchBuilds(project, null);
+  }
+
+  selectBranch = (branch) => {
+    this.fetchBuilds(this.props.project, branch.length > 0 ? branch : null);
+  }
+
+  render() {
+    const { apiKey } = this.props;
+    if (!apiKey) {
+      return (
+        <Redirect
+          to='/login'
+        />
+      )
+    }
+    return (
+      <div className="circle-ci-build-list">
+        <div className="row">
+          <div className="col-12 text-center">
+            <h1>Circle CI</h1>
+          </div>
+        </div>
+        <div className="row">
+          <div className="col-12">
+            <h4>Latest Builds</h4>
+          </div>
+        </div>
+        <div className="row">
+          <div className="col-3">
+            <ProjectSelect onSelect={this.selectProject} />
+          </div>
+          <div className="col-9">
+            <BranchSelect onSelect={this.selectBranch} />
+          </div>
+        </div>
+
+        {
+          this.props.builds.map((build, index) => {
+            let statusClass = 'circleci-build-status';
+            if(build.status === 'fixed' || build.status === 'success' ) {
+              statusClass += ' success'
+            } else if (build.status === 'failed'){
+              statusClass += ' fail'
+            } else if (build.status === 'running'){
+              statusClass += ' running'
+            }
+            return (
+              <div className="row circleci-build" key={`${build.vcs_type}-${build.username}-${build.reponame}-${build.build_num}`}>
+                <div className="col-12">
+                  <Link
+                    onClick={this.selectBuild(build)}
+                    to={`/build/${build.vcs_type}/${build.username}/${build.reponame}/${build.build_num}`}
+                    className="circleci-build-link"
+                  >
+                    <div className="row">
+                      <div className="col-10 circleci-build-title">
+                        {build.branch}/{build.reponame} #{build.build_num}
+                      </div>
+                      <div className="col-2">
+                        <TimeAgo date={build.queued_at} />
```

Comment:
Using moment is more used approach, but this is okay as well as it's okay-rated library